io/zenoh-transport/src/manager.rs

```diff
@@ -477,10 +477,12 @@ impl TransportManager {
     }
 
     pub async fn close(&self) {
+        self.close_with_timeout(Duration::from_secs(10)).await;
+    }
+
+    pub async fn close_with_timeout(&self, timeout: Duration) {
```

Comment:
It's an `async` function, is this really needed? All `async` functions return a `Future` that can be `await`ed with a timeout `xxx.close().timeout(_).await`.

Reply:
Inside I pass this `timeout` to `self.task_controller.terminate_all_async` .

Reply:
Yes, but looking at [terminate_all_async implementation](https://github.com/ZettaScaleLabs/zenoh/blob/e73a89d3bab5d3960159ddfc0ef315407d2fdf7e/commons/zenoh-task/src/lib.rs#L117) I'm not sure it needs to have a timeout either...

Reply:
Oh, right, I missed that
I will refactor this